sui-execution/latest/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -290,7 +303,7 @@ mod checked {
                     let msg = "Expected a coin but got an non coin object".to_owned();
                     return Err(ExecutionError::new_with_source(e, msg));
                 };
-                let split_coins = amount_args
+                let split_coins: Vec<Value> = amount_args
```

Comment:
Somehow `split_coins` being passed as a reference to `trace_split_coins` below whose parameter type is `&[Value]` confuses Rust's type system here and forces the use of type annotation